src/backend/pipeline/cont_analyze.c

```diff
@@ -484,28 +485,44 @@ find_clock_timestamp_expr(Node *node, ContAnalyzeContext *context)
 	return raw_expression_tree_walker(node, find_clock_timestamp_expr, (void *) context);
 }
 
-static char *
-get_truncation_from_interval_expr(Node *node)
+/*
+ * get_step_size_interval_expr
+ *
+ * Returns a constant representing the step size to use for the given window.
+ */
+static A_Const *
+get_step_size_interval_expr(Node *node)
 {
 	Expr *expr = (Expr *) transformExpr(make_parsestate(NULL), node, EXPR_KIND_WHERE);
-	Const *c;
-	Interval *i;
+	Const *window;
+	Interval *step;
+	Interval min_step;
+	float8 factor = (float8) sliding_window_step_factor / (float8) 100.0;
+	A_Const *result = makeNode(A_Const);
+	Value *v;
+	char *str;
+	Datum cmp;
+
+	MemSet(&min_step, 0, sizeof(Interval));
+	min_step.time = 1000 * 1000;
 
 	Assert(IsA(expr, Const));
-	c = (Const *) expr;
-	Assert(c->consttype == INTERVALOID);
+	window = (Const *) expr;
+	Assert(window->consttype == INTERVALOID);
+
+	/* We make the step size some percentage of the total window width */
+	step = (Interval *) DirectFunctionCall2(interval_mul, window->constvalue, Float8GetDatum(factor));
```

Comment:
Not sure if this will always work since `date_floor` only works when the interval size spans a single unit. So if the step here was `INTERVAL '75 second'`, `date_floor` would fail.